src/python/pants/backend/go/util_rules/external_pkg.py

```diff
@@ -251,40 +250,65 @@ class ExternalModulePkgImportPathsRequest:
     version: str
     go_mod_stripped_digest: Digest
 
-
-class ExternalModulePkgImportPaths(DeduplicatedCollection[str]):
-    """The import paths for all packages belonging to an external Go module."""
-
-    sort_input = True
+    def debug_hint(self) -> str:
+        return f"{self.module_path}@{self.version}"
 
 
 @rule
-async def compute_package_import_paths_from_external_module(
-    request: ExternalModulePkgImportPathsRequest,
-) -> ExternalModulePkgImportPaths:
+async def compute_external_module_metadata(
+    request: ExternalModuleInfoRequest,
+) -> ExternalModuleInfo:
     downloaded_module = await Get(
-        DownloadedModule,
-        DownloadedModuleRequest(
+        _DownloadedModule,
+        _DownloadedModuleRequest(
             request.module_path, request.version, request.go_mod_stripped_digest
         ),
     )
     json_result = await Get(
         ProcessResult,
         GoSdkProcess(
             input_digest=downloaded_module.digest,
-            # "-find" skips determining dependencies and imports for each package.
-            command=("list", "-find", "-mod=readonly", "-json", "./..."),
+            command=("list", "-mod=readonly", "-json", "./..."),
             env={"GOPROXY": "off"},
             description=(
-                "Determine packages belonging to Go external module "
-                f"{request.module_path}@{request.version}"
+                f"Determine metadata for Go external module {request.module_path}@{request.version}"
             ),
         ),
     )
-    return ExternalModulePkgImportPaths(
-        metadata["ImportPath"]
-        for metadata in ijson.items(json_result.stdout, "", multiple_values=True)
+
+    import_path_to_info = {}
+    for metadata in ijson.items(json_result.stdout, "", multiple_values=True):
+        import_path = metadata["ImportPath"]
+        pkg_info = ExternalPkgInfo(
```

Comment:
`ExternalPkgInfo` is going to need `HFiles` for assembly which I did not implement including in the assembly step. It would be prudent to also check if any of the other source keys are set (e.g., `CgoFiles`) and error if they are, so we don't try an incomplete build of an external package. This can be addressd as a separate PR.